src/main/java/jenkins/scm/api/SCMSource.java

```diff
@@ -261,6 +262,52 @@ protected SCMRevision retrieve(@NonNull SCMHead head, @NonNull TaskListener list
     }
 
     /**
+     * Looks up a specific revision based on some SCM-specific set of permissible syntaxes.
+     * Delegates to {@link #retrieve(String, TaskListener)}.
+     * @param revision might be a branch name, a tag name, a cryptographic hash, a revision number, etc.
+     * @param listener the task listener (optional)
+     * @return a valid revision object corresponding to the argument, with a usable corresponding head, or null if malformed or not found
+     * @throws IOException if an error occurs while performing the operation.
+     * @throws InterruptedException if any thread has interrupted the current thread.
+     * @since FIXME
+     */
+    @CheckForNull
+    public final SCMRevision fetch(@NonNull String revision, @CheckForNull TaskListener listener)
+            throws IOException, InterruptedException {
```

Comment:
What if the revision format is invalid?  We will end up with an IOException, but would it make sense to throw a more specific exception type than that (maybe InvalidSCMRevision)? 

This would also let us do some prevalidation before issuing network calls to we don't try with an invalid (or potentially security-compromising) revision specifier.